File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math
def solution(progresses, speeds):
    answer = []
    ist = []
    for i in range(0, len(progresses)):
        num = math.ceil((100-progresses[i])/speeds[i])

        if len(ist) == 0:
            ist.append(num)
        else :
            if ist[0]>= num:
                ist.append(num)
            else:
                answer.append(len(ist))
                ist = []
                ist.append(num)
    if len(ist) != 0:
        answer.append(len(ist))
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
class Solution {
    public List<Integer> solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<Integer>();
        List<Integer> time = new ArrayList<Integer>();
        Integer num = 0;
        
        for(int i = 0; i < progresses.length; i++){
            num = (int)Math.ceil((100-progresses[i]) / (double)speeds[i]);
            
            if(time.size() == 0){
                time.add(num);
            }
            
            else{
                
                if(time.get(0) < num){
                    answer.add(time.size());
                    time = new ArrayList<Integer>();
                    time.add(num);
                }
                else{
                    time.add(num);
                }
            }
        }
        if(time.size() != 0){
            answer.add(time.size());
        }
        return answer;
    }
}